### ISSUE TITLE
when I make after cmake,have 92 errors

### ISSUE THREAD

**Issue by gjhhust** [limbo018/DREAMPlace#113]
![image](https://user-images.githubusercontent.com/73829930/225781196-77e8e2e9-a383-4fb5-acfa-49ed99594a65.png)
according to the image,please hlep me,thanks

**Comment by gjhhust** [comment]
Thank your reply！！This problem does not exist on my new device, but there are new problems: My new System: Ubuntu 18, Torch 1.17.0, CUDA: 10.2, GCC: 7.5.0,GPU Capacity：6.1&nbsp; 。and this problem has been fixed after I changed the machine, when i run“python dreamplace/Placer.py test/ispd2005/adaptec1.json”，the new error is

And I looked up the issue and found two similar problems, one is not running in the root directory, and one is remake, but I tried and failed, what do you suggest




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Yibo ***@***.***&gt;; 
发送时间: 2023年3月17日(星期五) 晚上9:22
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [limbo018/DREAMPlace] when I make after cmake,have 92 errors (Issue #113)





 
Can you provide more information about your system, pytorch version, cuda version, and gcc version? The problem usually comes from incompatible compiling environments.

**Comment by limbo018** [comment]
It looks like you have solved the compiling problem. When running the program, you should run from the **install** direction, not the root direction, as mentioned in the readme.

**Comment by limbo018** [comment]
What problem? Can you upload the screenshot? I don't see any figures in your reply.

**Comment by gjhhust** [comment]
Oh!I'm sorry, I found that just failed to upload，here：
![G`EGGXS02QW@O}33$A @$O](https://user-images.githubusercontent.com/73829930/225921362-f97677d3-22ca-4f29-b7f1-8fe4660029ab.png)

**Comment by gjhhust** [comment]
yes, they are all environment set up under conda with Python 3.8.3.

**Comment by limbo018** [comment]
The problem probably comes from the linking with PyTorch, but I don't have much clue with the current information if you make sure that you compiled and ran the program under the same environment. I assume your PyTorch version is 1.7, as no version 1.17 has ever been released. 

I can share my environment for testing. 
```
Ubuntu 18.04
Python 3.8.10
gcc 7.5.0
PyTorch 1.7.1
CUDA 11.5
```
Your current problem comes from CPU version, so it has nothing to do with CUDA yet.

**Comment by limbo018** [comment]
Can you try only using the following command for building?
```
rm -r * # remove everything in build
cmake .. -DCMAKE_INSTALL_PREFIX=../../dreamplace_
```

**Comment by gjhhust** [comment]
The good news is that link succeeded, but unfortunately the data was out of bounds when i run:
![image](https://user-images.githubusercontent.com/73829930/225944571-c9e244a1-9a4f-4941-ac78-69255de4438e.png)
i try debug code：
![image](https://user-images.githubusercontent.com/73829930/225950897-176461b8-7a5e-4e66-b1cf-10c59da6140e.png)

**Comment by limbo018** [comment]
Hi,&nbsp;


You may try two directions:&nbsp;
1. Switch to PyTorch 1.7, as it has been tested already&nbsp;
2. Switch to the develop branch where I have merged a pull request which claims to fix the problem with PyTorch 2.0.&nbsp;




Sent&nbsp;from&nbsp;my&nbsp;iPhone


------------------ Original ------------------
From: gjhhust ***@***.***&gt;
Date: Fri, Mar 17, 2023 11:36 PM
To: limbo018/DREAMPlace ***@***.***&gt;
Cc: Yibo Lin ***@***.***&gt;, Comment ***@***.***&gt;
Subject: Re: [limbo018/DREAMPlace] when I make after cmake,have 92 errors(Issue #113)





 
The good news is that link succeeded, but unfortunately the data was out of bounds when i run:
 
 i try debug code：

**Comment by gjhhust** [comment]
Thank you! All the mistakes have been solved, thank you very much for your help, I wish you good health, Happy Life ~
